frontend/packages/topology/src/components/utils/checkResourceQuota.tsx

```diff
@@ -0,0 +1,37 @@
+import { getUsedPercentage } from '@console/app/src/components/resource-quota/utils';
+import { ResourceQuotaModel } from '../../../../../public/models';
+
+export const checkQuotaLimit = (resourecQuota: any) => {
```

Comment:
can we have type here? and as it's a list so can have it as plural `resourecQuotas`

and will be good to add unit test for `checkQuotaLimit`.

Reply:
Here `resourecQuotas ` will be either `ClusterResourceQuotaKind` or `ResourceQuotaKind`, I tried to add union of these two types but getting below error

`Each member of the union type '(<U>(callbackfn: (value: ClusterResourceQuotaKindProps, index: number, array: ClusterResourceQuotaKindProps[]) => U, thisArg?: any) => U[]) | (<U>(callbackfn: (value: ResourceQuotaKindProps, index: number, array: ResourceQuotaKindProps[]) => U, thisArg?: any) => U[])' has signatures, but none of those signatures are compatible with each other.`

When I check about the error, array.map does not correctly handles the union types . https://stackoverflow.com/questions/49510832/how-to-map-over-union-array-type 

I tried the solution provided but error still remains. 

Any feedback on this will be helpful.